### DISCUSSION TITLE
Importing interfaces from NPM; a centralised entity

### DISCUSSION THREAD

**Discussion by spencerj411** [Cyfrin/foundry-full-course-cu#2131] [General]
Hi, Cyfrin community! I'm currently on section 3 of solidity smart contract development course learning about how we can import smart contract interfaces from npm. My question is: how come we don't care that the interfaces we are importing are from a centralised entity; how come it does not matter? I am new but I imagine there is a risk of runtime errors? Thanks, all!

**Comment by EngrPips** [comment]
Hello @spencerj411, There is a level to which you can decentralize things, and some things don't even need to be decentralized. Blockchain as a system that allows decentralization of power and authority so no single entity is controlling the affairs of everybody makes sense but what is there to decentralize in a code everyone needs to use in their contract and which they can easily verify if it is doing what they expect it to do or not.

**Reply by spencerj411** [reply]
Hi @EngrPips, thanks for the prompt response!

A possibility is that an interface is changed (malicious or not), which could cause your deployed contract a runtime error (more of a concern if on mainnet). Is this not a valid concern?

Although, I can also imagine that if you're using a well-known interface, the code would be open-source and would have a well-designed version control behind it.

**Reply by EngrPips** [reply]
Every code we import in our codebase is always open source. Maybe Patrick made us use some code that is not open source, and I didn't catch it. Please point them out if you see any.
